public/app/core/components/NavBar/NavBarMenuItem.tsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import { GrafanaTheme2 } from '@grafana/data';
+import { Icon, IconName, Link, useTheme2 } from '@grafana/ui';
+import { css } from '@emotion/css';
+
+export interface Props {
+  icon?: IconName;
+  isActive?: boolean;
+  isDivider?: boolean;
+  isSectionHeader?: boolean;
+  onClick?: () => void;
+  target?: HTMLAnchorElement['target'];
+  text: string;
+  url?: string;
+}
+
+export function NavBarMenuItem({ icon, isActive, isDivider, isSectionHeader, onClick, target, text, url }: Props) {
+  const theme = useTheme2();
+  const styles = getStyles(theme, isActive, isSectionHeader);
+
+  const linkContent = (
+    <div className={styles.linkContent}>
+      <div>
+        {icon && <Icon data-testid="dropdown-child-icon" name={icon} className={styles.icon} />}
+        {text}
+      </div>
+      {target === '_blank' && (
+        <Icon data-testid="external-link-icon" name="external-link-alt" className={styles.externalLinkIcon} />
+      )}
+    </div>
+  );
+
+  let element = (
+    <button className={styles.element} onClick={onClick}>
+      {linkContent}
+    </button>
+  );
+
+  if (url) {
+    element =
+      !target && url.startsWith('/') ? (
+        <Link className={styles.element} href={url} target={target} onClick={onClick}>
+          {linkContent}
+        </Link>
+      ) : (
+        <a href={url} target={target} className={styles.element} onClick={onClick}>
+          {linkContent}
+        </a>
+      );
+  }
+
+  return isDivider ? (
+    <li data-testid="dropdown-child-divider" className={styles.divider} />
+  ) : (
+    <li className={styles.container}>{element}</li>
+  );
```

Comment:
Do the styles differ between mobile/desktop versions of NavBarMenuItem? If that is the case should we also create a `MobileNavBarMenuItem`?

Reply:
they do, but that's only because `NavBarDropdown` has the `dropdown-menu` and `dropdown-menu--sidemenu` classes which override the styling... trying to get to a point where we can not use them at all (we're close!).

i originally had 2 separate components. you're right the styling is slightly different, but the logic inside the component ends up being the same. wonder whether we should make this component even more generic, and provide a way for the consuming component to apply additional styling, e.g. add a `className` prop that gets merged into the top level container class... wdyt @hugohaggmark? 🤔

Reply:
I was thinking about the same, i.e. introduce a style prop, as long as the generic component doesn't get too complex.